scripts/generateExampleCertificates.ts

```diff
@@ -0,0 +1,291 @@
+const testScopeKey = '@test/app';
+
+// generated with oidgen script. in the microsoft OID space. could apply for Expo space but would take time: https://pen.iana.org/pen/PenApplication.page
+const expoProjectInformationOID =
+  '1.2.840.113556.1.8000.2554.43437.254.128.102.157.7894389.20439.2.1';
+
+export async function run(): Promise<void> {
+  const root = await generateExpoRootCertificateAsync();
+  await exportCertificateAndKeysAsync(root, 'expo-root');
+
+  const expoGo = await generateExpoGoIntermediateCertificate(root);
+  await exportCertificateAndKeysAsync(expoGo, 'expo-go');
+
+  const developmentCSR = await generateDevelopmentCSR(testAppId, testScopeKey);
+
+  const testDevelopmentCert = await generateTestDevelopmentCertificate(developmentCSR, expoGo);
+  await exportCertificateAndKeysAsync(testDevelopmentCert, 'development');
+
+  // verify certificate chain
+  // const caStore = PKI.createCaStore();
+  // caStore.addCertificate(expoGo.certificate);
+  // const chain = [testDevelopmentCert.certificate, expoGo.certificate];
+  // PKI.verifyCertificateChain(caStore, chain);
+}
+
+async function exportCertificateAndKeysAsync(
+  { privateKey, publicKey, certificate }: KeysAndCertificate,
+  prefix: string
+): Promise<void> {
+  await Promise.all([
+    fs.writeFile(`keys/${prefix}-public-key.pem`, PKI.publicKeyToPem(publicKey)),
+    fs.writeFile(`keys/${prefix}-private-key.pem`, PKI.privateKeyToPem(privateKey)),
+    fs.writeFile(`keys/${prefix}-certificate.pem`, PKI.certificateToPem(certificate)),
+  ]);
+}
+
+async function generateExpoRootCertificateAsync(): Promise<KeysAndCertificate> {
+  const { privateKey, publicKey } = generateKeyPair();
+
+  const certificate = PKI.createCertificate();
+  certificate.publicKey = publicKey;
+  certificate.serialNumber = toPositiveHex(util.bytesToHex(random.getBytesSync(9)));
+
+  // 20 year validity
+  certificate.validity.notBefore = new Date();
+  certificate.validity.notAfter = new Date();
+  certificate.validity.notAfter.setFullYear(certificate.validity.notBefore.getFullYear() + 20);
+
+  const attrs = [
+    {
+      name: 'commonName',
+      value: 'Expo Root Certificate',
+    },
+    {
+      name: 'countryName',
+      value: 'US',
+    },
+    {
+      shortName: 'ST',
+      value: 'California',
+    },
+    {
+      name: 'localityName',
+      value: 'Palo Alto',
+    },
+    {
+      name: 'organizationName',
+      value: 'Expo',
+    },
+    {
+      shortName: 'OU',
+      value: 'Engineering',
+    },
+    {
+      name: 'emailAddress',
+      value: 'secure@expo.dev',
+    },
+  ];
+
+  // root certificate is self-signed
+  certificate.setSubject(attrs);
+  certificate.setIssuer(attrs);
+
+  certificate.setExtensions([
+    {
+      name: 'basicConstraints',
+      cA: true,
+      pathLenConstraint: 1, // should be at most 1 intermediate certificate (expo go intermediate CA + development certs issued by that intermediate CA)
```

Comment:
IMO we should leave this off for the root since it's hard to rotate this certificate. DigiCert's root CA has no path length for example